### PR TITLE
QRadar - improvment in mirroring performance

### PR DESCRIPTION
**Pull request by ilaner**
**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/XSUP-18027?filter=-1)
fixes:  https://jira-hq.paloaltonetworks.local/browse/XSUP-19419?focusedCommentId=6194265&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-6194265

## Description
Major improvement in QRadar mirroring
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py [hunk 1]
```diff
@@ -1897,12 +1907,12 @@ def get_incidents_long_running_execution(client: Client, offenses_per_fetch: int
     return incidents, new_highest_offense_id
 
 
-def prepare_context_for_failed_events(offenses_with_success):
+def prepare_context_for_events(offenses_with_metadata):
     ctx, version = get_integration_context_with_version()
     changed_offense_ids = []
-    for offense, is_success in offenses_with_success:
+    for offense, is_success in offenses_with_metadata:
+        offense_id = str(offense['id'])
```
Comment (tkatzir) [inline Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py:1914]: Why move it outside the `if`?

File: Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py [hunk 2]
```diff
@@ -3484,38 +3496,48 @@ def add_modified_remote_offenses(client: Client,
...
                 new_modified_records_ids.add(offense_id)
                 changed_ids_ctx.append(offense_id)
+                current_concurrent_searches -= 1
             else:
                 print_debug_msg(f'offense {offense_id}, search query {search_id}, status is {status}')
-            time.sleep(EVENTS_MODIFIED_SECS)
+
+        for offense_id in offense_ids_to_search:
+            if current_concurrent_searches >= MAX_SEARCHES_QUEUE:
+                break
```
Comment (tkatzir) [inline Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py:3527]: Add (info?) log message that MAX_SEARCHES was reached, and not creating another search.

**Review by tkatzir — COMMENTED** [review]
Good additions. Have some comments.

File: Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py (hunk 2 above)
Comment (ilaner) [reply Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py:3541]: added

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good, seems comments.

**Review by DeanArbel — APPROVED** [review]

**Pull request merged**